cypress/component/integration/Button.cy.tsx

```diff
@@ -0,0 +1,342 @@
+        cy.mount(
+          <div className="theme-dark">
+            <Button preIcon={<FiArrowUpRight data-testid="preIcon" />} />;
+          </div>,
+        );
+      });
+
+      it('should render primary button with preIcon', () => {
+        cy.get('[data-testid="button"]').should('exist');
+        cy.get('[data-testid="preIcon"]').should('exist');
+        cy.get('[data-testid="posIcon"]').should('not.exist');
+      });
+
+      it('should match snapshot', () => {
+        compareSnapshot(cy, 'primary-button-with-preIcon');
+      });
+    });
+
+    describe('primary button with posIcon', () => {
+      beforeEach(() => {
+        cy.mount(
+          <div className="theme-dark">
+            <Button posIcon={<FiArrowUpRight data-testid="posIcon" />} />
+          </div>,
+        );
+      });
+      it('should render primary button with posIcon', () => {
+        cy.get('[data-testid="button"]').should('exist');
+        cy.get('[data-testid="preIcon"]').should('not.exist');
+        cy.get('[data-testid="posIcon"]').should('exist');
+      });
+
+      it('should match snapshot', () => {
+        compareSnapshot(cy, 'primary-button-with-posIcon');
+      });
+    });
+
+    describe('primary button with both icons and with children', () => {
+      beforeEach(() => {
+        cy.mount(
+          <div className="theme-dark">
+            <Button
+              preIcon={<FiArrowUpRight data-testid="preIcon" />}
+              posIcon={<FiArrowUpRight data-testid="posIcon" />}
+            >
+              <div data-testid="children">Children</div>
+            </Button>
+            ,
+          </div>,
+        );
+      });
+      it('should render primary button with both icons and children', () => {
+        cy.get('[data-testid="button"]').should('exist');
+        cy.get('[data-testid="preIcon"]').should('exist');
+        cy.get('[data-testid="posIcon"]').should('exist');
+        cy.get('[data-testid="children"]').should('exist');
+        cy.get('[data-testid="children"]').should('contain', 'Children');
+      });
+
+      it('should match snapshot', () => {
+        compareSnapshot(cy, 'primary-button-with-both-icons-and-with-children');
+      });
+    });
+
+    describe('primary button with preIcon with children', () => {
+      beforeEach(() => {
+        cy.mount(
+          <div className="theme-dark">
+            <Button preIcon={<FiArrowUpRight data-testid="preIcon" />}>
+              <div data-testid="children">Children</div>
+            </Button>
+            ,
+          </div>,
+        );
+      });
+      it('should render primary button with preIcon and with children', () => {
+        cy.get('[data-testid="button"]').should('exist');
+        cy.get('[data-testid="preIcon"]').should('exist');
+        cy.get('[data-testid="posIcon"]').should('not.exist');
+        cy.get('[data-testid="children"]').should('exist');
+        cy.get('[data-testid="children"]').should('contain', 'Children');
```

Comment:
Assert `contain` is also valid for `exist`. I would keep only one.

```suggestion
        cy.get('[data-testid="children"]').should('contain', 'Children');
```